--- libgcrypt-1.5.0.org/src/gcrypt.h.in	2011-06-29 10:58:56.000000000 +0200
+++ libgcrypt-1.5.0/src/gcrypt.h.in	2013-01-02 18:00:50.534124024 +0100
@@ -1249,6 +1249,7 @@
 gcry_error_t gcry_md_list (int *list, int *list_length);
 
 
+#if !defined(GCRYPT_NO_DEPRECATED) || defined(_GCRYPT_IN_LIBGCRYPT) 
 /* Alternative interface for asymmetric cryptography.  This interface
    is deprecated.  */
 
@@ -1400,7 +1401,7 @@
 {
   gcry_md_algo_t md;
 } gcry_ac_ssa_pkcs_v1_5_t _GCRY_ATTR_INTERNAL;
-
+#endif
 
 #ifndef GCRYPT_NO_DEPRECATED
 /* Returns a new, empty data set in DATA.  */
