diff -Naur xdelta3-3.1.0.org/testing/checksum_test.cc xdelta3-3.1.0/testing/checksum_test.cc
--- xdelta3-3.1.0.org/testing/checksum_test.cc	2016-01-01 07:34:33.000000000 +0100
+++ xdelta3-3.1.0/testing/checksum_test.cc	2026-06-28 16:21:58.606427800 +0200
@@ -453,12 +453,12 @@
 
   void print() {
     if (fstats.count != count()) {
-      fprintf(stderr, "internal error: %" W "d != %" W "d\n", fstats.count, count());
+      fprintf(stderr, "internal error: %" XD3_W "d != %" XD3_W "d\n", fstats.count, count());
       abort();
     }
     print_header();
-    printf("%-32s%d/%d 2^%" W "u\t%" W "u\t%0.4f\t%.4f\t%.4f\t%.1e\t%.2f\t"
-	   "%" W "u\t%" W "u\n",
+    printf("%-32s%d/%d 2^%" XD3_W "u\t%" XD3_W "u\t%0.4f\t%.4f\t%.4f\t%.1e\t%.2f\t"
+	   "%" XD3_W "u\t%" XD3_W "u\n",
 	   test_name,
 	   Checksum::cksum_size,
 	   Checksum::cksum_skip,
diff -Naur xdelta3-3.1.0.org/testing/file.h xdelta3-3.1.0/testing/file.h
--- xdelta3-3.1.0.org/testing/file.h	2016-01-08 07:49:10.000000000 +0100
+++ xdelta3-3.1.0/testing/file.h	2026-06-28 16:21:58.607427812 +0200
@@ -316,7 +316,7 @@
     pid_t pid = getpid();
     char buf[64];
     xoff_t xpid = pid;
-    snprintf(buf, 64, "/tmp/regtest.%" Q "u.%d", xpid, static_counter++);
+    snprintf(buf, 64, "/tmp/regtest.%" XD3_Q "u.%d", xpid, static_counter++);
     filename_.append(buf);
     unlink(filename_.c_str());
   }
diff -Naur xdelta3-3.1.0.org/testing/regtest.cc xdelta3-3.1.0/testing/regtest.cc
--- xdelta3-3.1.0.org/testing/regtest.cc	2016-01-05 09:09:40.000000000 +0100
+++ xdelta3-3.1.0/testing/regtest.cc	2026-06-28 16:21:58.607427812 +0200
@@ -92,7 +92,7 @@
     bool done = false;
     bool done_after_input = false;
 
-    IF_DEBUG1 (XPR(NTR "source %" Q "u[%" Z "u] target %" Q "u winsize %" Z "u\n",
+    IF_DEBUG1 (XPR(NTR "source %" XD3_Q "u[%" XD3_Z "u] target %" XD3_Q "u winsize %" XD3_Z "u\n",
 		   source_file.Size(), options.block_size,
 		   target_file.Size(),
 		   Constants::WINDOW_SIZE));
@@ -102,8 +102,8 @@
 
       xoff_t blks = target_iterator.Blocks();
 
-      IF_DEBUG2(XPR(NTR "target in %s: %" Q "u[%" Z "u] %" Q "u(%" Q "u) "
-		    "verified %" Q "u\n",
+      IF_DEBUG2(XPR(NTR "target in %s: %" XD3_Q "u[%" XD3_Z "u] %" XD3_Q "u(%" XD3_Q "u) "
+		    "verified %" XD3_Q "u\n",
 		    encoding ? "encoding" : "decoding",
 		    target_iterator.Offset(),
 		    target_block.Size(),
@@ -155,8 +155,8 @@
 	xd3_source *src = (encoding ? &encode_source : &decode_source);
 	Block *block = (encoding ? &encode_source_block : &decode_source_block);
 	if (encoding) {
-	  IF_DEBUG2(XPR(NTR "[srcblock] %" Q "u last srcpos %" Q "u "
-			"encodepos %" Q "u\n",
+	  IF_DEBUG2(XPR(NTR "[srcblock] %" XD3_Q "u last srcpos %" XD3_Q "u "
+			"encodepos %" XD3_Q "u\n",
 			encode_source.getblkno,
 			encode_stream.match_last_srcpos,
 			encode_stream.input_position + encode_stream.total_in));
@@ -233,7 +233,7 @@
 			const Options &options) {
     vector<const char*> ecmd;
     char bbuf[16];
-    snprintf(bbuf, sizeof(bbuf), "-B%" Q "u", options.encode_srcwin_maxsz);
+    snprintf(bbuf, sizeof(bbuf), "-B%" XD3_Q "u", options.encode_srcwin_maxsz);
     ecmd.push_back("xdelta3");
     ecmd.push_back(bbuf);
     ecmd.push_back("-s");
@@ -373,7 +373,7 @@
 void TestPrintf() {
   char buf[64];
   xoff_t x = XOFF_T_MAX;
-  snprintf_func (buf, sizeof(buf), "%" Q "u", x);
+  snprintf_func (buf, sizeof(buf), "%" XD3_Q "u", x);
   const char *expect = XD3_USE_LARGEFILE64 ?
     "18446744073709551615" : "4294967295";
   XD3_ASSERT(strcmp (buf, expect) == 0);
@@ -816,7 +816,7 @@
     InMemoryEncodeDecode(spec0, spec1, &block, options);
     Delta delta(block);
 
-    IF_DEBUG1(DP(RINT "[stride=%d] changes=%" W "u adds=%" Q "u\n",
+    IF_DEBUG1(DP(RINT "[stride=%d] changes=%" XD3_W "u adds=%" XD3_Q "u\n",
 		 s, changes, delta.AddedBytes()));
     double allowance = Constants::BLOCK_SIZE < 8192 || s < 30 ? 3.0 : 1.1;
     CHECK_GE(allowance * changes, (double)delta.AddedBytes());
@@ -1262,7 +1262,7 @@
 // These are Xdelta tests.
 template <class T>
 void MainTest() {
-  XPR(NT "Blocksize %" Q "u windowsize %" Z "u\n",
+  XPR(NT "Blocksize %" XD3_Q "u windowsize %" XD3_Z "u\n",
       T::BLOCK_SIZE, T::WINDOW_SIZE);
   Regtest<T> regtest;
   TEST(TestEmptyInMemory);
diff -Naur xdelta3-3.1.0.org/xdelta3-blkcache.h xdelta3-3.1.0/xdelta3-blkcache.h
--- xdelta3-3.1.0.org/xdelta3-blkcache.h	2016-01-05 08:41:09.000000000 +0100
+++ xdelta3-3.1.0/xdelta3-blkcache.h	2026-06-28 16:21:58.607427812 +0200
@@ -248,7 +248,7 @@
 
       if (sfile->size_known)
 	{
-	  short_sprintf (srcszbuf, "source size %s [%"Q"u]",
+	  short_sprintf (srcszbuf, "source size %s [%"XD3_Q"u]",
 			 main_format_bcnt (source_size, &srccntbuf),
 			 source_size);
 	}
@@ -261,7 +261,7 @@
 
       if (option_verbose > 1)
 	{
-	  short_sprintf (nbufs, " #bufs %"W"u", lru_size);
+	  short_sprintf (nbufs, " #bufs %"XD3_W"u", lru_size);
 	}
 
       XPR(NT "source %s %s blksize %s window %s%s%s\n",
@@ -372,7 +372,7 @@
 	  if (!option_quiet)
 	    {
 	      XPR(NT "source can't seek backwards; requested block offset "
-		  "%"Q"u source position is %"Q"u\n",
+		  "%"XD3_Q"u source position is %"XD3_Q"u\n",
 		  pos, sfile->source_position);
 	    }
 
@@ -392,7 +392,7 @@
 
 	  if (option_verbose > 1)
 	    {
-	      XPR(NT "seek error at offset %"Q"u: %s\n",
+	      XPR(NT "seek error at offset %"XD3_Q"u: %s\n",
 		  pos, xd3_mainerror (ret));
 	    }
 	}
@@ -401,7 +401,7 @@
 
       if (option_verbose > 1 && pos != sfile->source_position)
 	{
-	  XPR(NT "non-seekable source skipping %"Q"u bytes @ %"Q"u\n",
+	  XPR(NT "non-seekable source skipping %"XD3_Q"u bytes @ %"XD3_Q"u\n",
 	      pos - sfile->source_position,
 	      sfile->source_position);
 	}
@@ -531,20 +531,20 @@
 	{
 	  if (blru->blkno != blkno)
 	    {
-	      XPR(NT "source block %"Q"u read %"Z"u ejects %"Q"u (lru_hits=%u, "
+	      XPR(NT "source block %"XD3_Q"u read %"XD3_Z"u ejects %"XD3_Q"u (lru_hits=%u, "
 		  "lru_misses=%u, lru_filled=%u)\n",
 		  blkno, nread, blru->blkno, lru_hits, lru_misses, lru_filled);
 	    }
 	  else
 	    {
-	      XPR(NT "source block %"Q"u read %"Z"u (lru_hits=%u, "
+	      XPR(NT "source block %"XD3_Q"u read %"XD3_Z"u (lru_hits=%u, "
 		  "lru_misses=%u, lru_filled=%u)\n",
 		  blkno, nread, lru_hits, lru_misses, lru_filled);
 	    }
 	}
       else
 	{
-	  XPR(NT "source block %"Q"u read %"Z"u (lru_hits=%u, lru_misses=%u, "
+	  XPR(NT "source block %"XD3_Q"u read %"XD3_Z"u (lru_hits=%u, lru_misses=%u, "
 	      "lru_filled=%u)\n", blkno, nread, 
 	      lru_hits, lru_misses, lru_filled);
 	}
diff -Naur xdelta3-3.1.0.org/xdelta3-main.h xdelta3-3.1.0/xdelta3-main.h
--- xdelta3-3.1.0.org/xdelta3-main.h	2016-01-07 07:37:25.000000000 +0100
+++ xdelta3-3.1.0/xdelta3-main.h	2026-06-28 16:21:58.608427825 +0200
@@ -604,7 +604,7 @@
 
       if (r >= 100 && r < 1000)
 	{
-	  short_sprintf (*buf, "%"Q"u %s", r, fmts[i]);
+	  short_sprintf (*buf, "%"XD3_Q"u %s", r, fmts[i]);
 	  return buf->buf;
 	}
 
@@ -707,12 +707,12 @@
 
   if (x < low)
     {
-      XPR(NT "-%c: minimum value: %"Q"u\n", which, low);
+      XPR(NT "-%c: minimum value: %"XD3_Q"u\n", which, low);
       return EXIT_FAILURE;
     }
   if (high != 0 && x > high)
     {
-      XPR(NT "-%c: maximum value: %"Q"u\n", which, high);
+      XPR(NT "-%c: maximum value: %"XD3_Q"u\n", which, high);
       return EXIT_FAILURE;
     }
   (*xo) = x;
@@ -1043,7 +1043,7 @@
 		const char *msg)
 {
   int ret = 0;
-  IF_DEBUG1(DP(RINT "[main] read %s up to %"Z"u\n", ifile->filename, size));
+  IF_DEBUG1(DP(RINT "[main] read %s up to %"XD3_Z"u\n", ifile->filename, size));
 
 #if XD3_STDIO
   size_t result;
@@ -1071,7 +1071,7 @@
     }
   else
     {
-      if (option_verbose > 4) { XPR(NT "read %s: %"Z"u bytes\n",
+      if (option_verbose > 4) { XPR(NT "read %s: %"XD3_Z"u bytes\n",
 				    ifile->filename, (*nread)); }
       ifile->nread += (*nread);
     }
@@ -1107,7 +1107,7 @@
     }
   else
     {
-      if (option_verbose > 5) { XPR(NT "write %s: %"W"u bytes\n",
+      if (option_verbose > 5) { XPR(NT "write %s: %"XD3_W"u bytes\n",
 				    ofile->filename, size); }
       ofile->nwrite += size;
     }
@@ -1306,7 +1306,7 @@
 
       if ((ret = xd3_decode_instruction (stream)))
 	{
-	  XPR(NT "instruction decode error at %"Q"u: %s\n",
+	  XPR(NT "instruction decode error at %"XD3_Q"u: %s\n",
 	      stream->dec_winstart + size, stream->msg);
 	  return ret;
 	}
@@ -1314,7 +1314,7 @@
       addr_bytes = (usize_t)(stream->addr_sect.buf - addr_before);
       inst_bytes = (usize_t)(stream->inst_sect.buf - inst_before);
 
-      VC(UT "  %06"Q"u %03"W"u  %s %6"W"u", 
+      VC(UT "  %06"XD3_Q"u %03"XD3_W"u  %s %6"XD3_W"u", 
 	 stream->dec_winstart + size,
 	 option_print_cpymode ? code : 0,
 	 xd3_rtype_to_string ((xd3_rtype) stream->dec_current1.type,
@@ -1327,12 +1327,12 @@
 	    {
 	      if (stream->dec_current1.addr >= stream->dec_cpylen)
 		{
-		  VC(UT " T@%-6"W"u",
+		  VC(UT " T@%-6"XD3_W"u",
 		     stream->dec_current1.addr - stream->dec_cpylen)VE;
 		}
 	      else
 		{
-		  VC(UT " S@%-6"Q"u",
+		  VC(UT " S@%-6"XD3_Q"u",
 		     stream->dec_cpyoff + stream->dec_current1.addr)VE;
 		}
 	    }
@@ -1346,7 +1346,7 @@
 
       if (stream->dec_current2.type != XD3_NOOP)
 	{
-	  VC(UT "  %s %6"W"u",
+	  VC(UT "  %s %6"XD3_W"u",
 	     xd3_rtype_to_string ((xd3_rtype) stream->dec_current2.type,
 				  option_print_cpymode),
 	     stream->dec_current2.size)VE;
@@ -1355,12 +1355,12 @@
 	    {
 	      if (stream->dec_current2.addr >= stream->dec_cpylen)
 		{
-		  VC(UT " T@%-6"W"u",
+		  VC(UT " T@%-6"XD3_W"u",
 		     stream->dec_current2.addr - stream->dec_cpylen)VE;
 		}
 	      else
 		{
-		  VC(UT " S@%-6"Q"u",
+		  VC(UT " S@%-6"XD3_Q"u",
 		     stream->dec_cpyoff + stream->dec_current2.addr)VE;
 		}
 	    }
@@ -1375,7 +1375,7 @@
 	  (stream->dec_current1.type >= XD3_CPY ||
 	   stream->dec_current2.type >= XD3_CPY))
 	{
-	  VC(UT "  %06"Q"u (inefficiency) %"W"u encoded as %"W"u bytes\n",
+	  VC(UT "  %06"XD3_Q"u (inefficiency) %"XD3_W"u encoded as %"XD3_W"u bytes\n",
 	     stream->dec_winstart + size_before,
 	     size - size_before,
 	     addr_bytes + inst_bytes)VE;
@@ -1443,7 +1443,7 @@
   if (stream->dec_winstart == 0)
     {
       VC(UT "VCDIFF version:               0\n")VE;
-      VC(UT "VCDIFF header size:           %"W"u\n",
+      VC(UT "VCDIFF header size:           %"XD3_W"u\n",
 	 stream->dec_hdrsize)VE;
       VC(UT "VCDIFF header indicator:      ")VE;
       if ((stream->dec_hdr_ind & VCD_SECONDARY) != 0)
@@ -1498,7 +1498,7 @@
       VC(UT "\n")VE;
     }
 
-  VC(UT "VCDIFF window number:         %"Q"u\n", stream->current_window)VE;
+  VC(UT "VCDIFF window number:         %"XD3_Q"u\n", stream->current_window)VE;
   VC(UT "VCDIFF window indicator:      ")VE;
   if ((stream->dec_win_ind & VCD_SOURCE) != 0) VC(UT "VCD_SOURCE ")VE;
   if ((stream->dec_win_ind & VCD_TARGET) != 0) VC(UT "VCD_TARGET ")VE;
@@ -1524,27 +1524,27 @@
 
   if (stream->dec_winstart != 0)
     {
-      VC(UT "VCDIFF window at offset:      %"Q"u\n", stream->dec_winstart)VE;
+      VC(UT "VCDIFF window at offset:      %"XD3_Q"u\n", stream->dec_winstart)VE;
     }
 
   if (SRCORTGT (stream->dec_win_ind))
     {
-      VC(UT "VCDIFF copy window length:    %"W"u\n",
+      VC(UT "VCDIFF copy window length:    %"XD3_W"u\n",
 	 stream->dec_cpylen)VE;
-      VC(UT "VCDIFF copy window offset:    %"Q"u\n",
+      VC(UT "VCDIFF copy window offset:    %"XD3_Q"u\n",
 	 stream->dec_cpyoff)VE;
     }
 
-  VC(UT "VCDIFF delta encoding length: %"W"u\n",
+  VC(UT "VCDIFF delta encoding length: %"XD3_W"u\n",
      (usize_t)stream->dec_enclen)VE;
-  VC(UT "VCDIFF target window length:  %"W"u\n",
+  VC(UT "VCDIFF target window length:  %"XD3_W"u\n",
      (usize_t)stream->dec_tgtlen)VE;
 
-  VC(UT "VCDIFF data section length:   %"W"u\n",
+  VC(UT "VCDIFF data section length:   %"XD3_W"u\n",
      (usize_t)stream->data_sect.size)VE;
-  VC(UT "VCDIFF inst section length:   %"W"u\n",
+  VC(UT "VCDIFF inst section length:   %"XD3_W"u\n",
      (usize_t)stream->inst_sect.size)VE;
-  VC(UT "VCDIFF addr section length:   %"W"u\n",
+  VC(UT "VCDIFF addr section length:   %"XD3_W"u\n",
      (usize_t)stream->addr_sect.size)VE;
 
   ret = 0;
@@ -2263,7 +2263,7 @@
 
   if (option_verbose && skipped != 0)
     {
-      XPR(NT "skipping %"Q"u bytes in %s\n",
+      XPR(NT "skipping %"XD3_Q"u bytes in %s\n",
 	  skipped, ifile->filename);
     }
   return 0;
@@ -3304,7 +3304,7 @@
 		    /* Warn when no source copies are found */
 		    if (option_verbose && ! xd3_encoder_used_source (& stream))
 		      {
-			XPR(NT "warning: input window %"Q"u..%"Q"u has "
+			XPR(NT "warning: input window %"XD3_Q"u..%"XD3_Q"u has "
 			    "no source copies\n",
 			    stream.current_window * winsize,
 			    (stream.current_window+1) * winsize);
@@ -3317,8 +3317,8 @@
 			stream.srcwin_decided_early &&
 			stream.i_slots_used > stream.iopt_size)
 		      {
-			XPR(NT "warning: input position %"Q"u overflowed "
-			    "instruction buffer, needed %"W"u (vs. %"W"u), "
+			XPR(NT "warning: input position %"XD3_Q"u overflowed "
+			    "instruction buffer, needed %"XD3_W"u (vs. %"XD3_W"u), "
 			    "consider changing -I\n",
 			    stream.current_window * winsize,
 			    stream.i_slots_used, stream.iopt_size);
@@ -3341,7 +3341,7 @@
 
 		    if (option_verbose > 1)
 		      {
-			XPR(NT "%"Q"u: in %s (%s): out %s (%s): "
+			XPR(NT "%"XD3_Q"u: in %s (%s): out %s (%s): "
 			    "total in %s: out %s: %s: srcpos %s\n",
 			    stream.current_window,
 			    main_format_bcnt (this_read, &rdb),
@@ -3355,7 +3355,7 @@
 		      }
 		    else
 		      {
-			XPR(NT "%"Q"u: in %s: out %s: total in %s: "
+			XPR(NT "%"XD3_Q"u: in %s: out %s: total in %s: "
 			    "out %s: %s\n",
  			    stream.current_window,
 			    main_format_bcnt (this_read, &rdb),
@@ -3443,21 +3443,21 @@
   if (option_verbose > 1 && cmd == CMD_ENCODE)
     {
       XPR(NT "scanner configuration: %s\n", stream.smatcher.name);
-      XPR(NT "target hash table size: %"W"u\n", stream.small_hash.size);
+      XPR(NT "target hash table size: %"XD3_W"u\n", stream.small_hash.size);
       if (sfile != NULL && sfile->filename != NULL)
 	{
-	  XPR(NT "source hash table size: %"W"u\n", stream.large_hash.size);
+	  XPR(NT "source hash table size: %"XD3_W"u\n", stream.large_hash.size);
 	}
     }
 
   if (option_verbose > 2 && cmd == CMD_ENCODE)
     {
-      XPR(NT "source copies: %"Q"u (%"Q"u bytes)\n",
+      XPR(NT "source copies: %"XD3_Q"u (%"XD3_Q"u bytes)\n",
 	  stream.n_scpy, stream.l_scpy);
-      XPR(NT "target copies: %"Q"u (%"Q"u bytes)\n",
+      XPR(NT "target copies: %"XD3_Q"u (%"XD3_Q"u bytes)\n",
 	  stream.n_tcpy, stream.l_tcpy);
-      XPR(NT "adds: %"Q"u (%"Q"u bytes)\n", stream.n_add, stream.l_add);
-      XPR(NT "runs: %"Q"u (%"Q"u bytes)\n", stream.n_run, stream.l_run);
+      XPR(NT "adds: %"XD3_Q"u (%"XD3_Q"u bytes)\n", stream.n_add, stream.l_add);
+      XPR(NT "runs: %"XD3_Q"u (%"XD3_Q"u bytes)\n", stream.n_run, stream.l_run);
     }
 #endif
 
@@ -3469,7 +3469,7 @@
       long end_time = get_millisecs_now ();
       xoff_t nwrite = ofile != NULL ? ofile->nwrite : 0;
 
-      XPR(NT "finished in %s; input %"Q"u output %"Q"u bytes (%0.2f%%)\n",
+      XPR(NT "finished in %s; input %"XD3_Q"u output %"XD3_Q"u bytes (%0.2f%%)\n",
 	  main_format_millis (end_time - start_time, &tm),
 	  ifile->nread, nwrite, 100.0 * nwrite / ifile->nread);
     }
diff -Naur xdelta3-3.1.0.org/xdelta3-test.h xdelta3-3.1.0/xdelta3-test.h
--- xdelta3-3.1.0.org/xdelta3-test.h	2016-01-05 08:07:04.000000000 +0100
+++ xdelta3-3.1.0/xdelta3-test.h	2026-06-28 16:21:58.615427914 +0200
@@ -223,7 +223,7 @@
 {
   char buf[64];
   xoff_t x = XOFF_T_MAX;
-  snprintf_func (buf, sizeof(buf), "%"Q"u", x);
+  snprintf_func (buf, sizeof(buf), "%"XD3_Q"u", x);
   const char *expect = XD3_USE_LARGEFILE64 ?
     "18446744073709551615" : "4294967295";
   if (strcmp (buf, expect) == 0) {
@@ -441,7 +441,7 @@
 	{
 	  if (obuf[i] != rbuf[i])
  	    {
-	      XPR(NT "byte %u (read %u @ %"Q"u) %d != %d\n",
+	      XPR(NT "byte %u (read %u @ %"XD3_Q"u) %d != %d\n",
 		  (int)i, (int)oc, offset, obuf[i], rbuf[i]);
 	      diffs++;
 	      return XD3_INTERNAL;
@@ -1425,7 +1425,7 @@
 	  if ((ret = sec->encode (stream, enc_stream,
 				  in_head, out_head, & cfg)))
 	    {
-	      XPR(NT "test %"W"u: encode: %s", test_i, stream->msg);
+	      XPR(NT "test %"XD3_W"u: encode: %s", test_i, stream->msg);
 	      goto fail;
 	    }
 
@@ -1464,7 +1464,7 @@
 					    compress_size, dec_input,
 					    dec_correct, dec_output)))
 	    {
-	      XPR(NT "test %"W"u: decode: %s", test_i, stream->msg);
+	      XPR(NT "test %"XD3_W"u: decode: %s", test_i, stream->msg);
 	      goto fail;
 	    }
 
@@ -1833,7 +1833,7 @@
       if (ratio >= TEST_ADD_RATIO + TEST_EPSILON)
 	{
 	  XPR(NT "test encode with size ratio %.4f, "
-	     "expected < %.4f (%"Q"u, %"Q"u)\n",
+	     "expected < %.4f (%"XD3_Q"u, %"XD3_Q"u)\n",
 	    ratio, TEST_ADD_RATIO + TEST_EPSILON, dsize, tsize);
 	  stream->msg = "strange encoding";
 	  return XD3_INTERNAL;
@@ -2760,14 +2760,14 @@
 	    default: CHECK(0);
 	    }
 
-	  snprintf_func (rptr, rbuf+TESTBUFSIZE-rptr, "%"W"u/%"W"u",
+	  snprintf_func (rptr, rbuf+TESTBUFSIZE-rptr, "%"XD3_W"u/%"XD3_W"u",
 			 inst->pos, inst->size);
 	  rptr += strlen (rptr);
 
 	  if (inst->type == XD3_CPY)
 	    {
 	      *rptr++ = '@';
-	      snprintf_func (rptr, rbuf+TESTBUFSIZE-rptr, "%"Q"u", inst->addr);
+	      snprintf_func (rptr, rbuf+TESTBUFSIZE-rptr, "%"XD3_Q"u", inst->addr);
 	      rptr += strlen (rptr);
 	    }
 
@@ -2783,7 +2783,7 @@
 
       if (strcmp (rbuf, test->result) != 0)
 	{
-	  XPR(NT "test %"W"u: expected %s: got %s", i, test->result, rbuf);
+	  XPR(NT "test %"XD3_W"u: expected %s: got %s", i, test->result, rbuf);
 	  stream->msg = "wrong result";
 	  return XD3_INTERNAL;
 	}
diff -Naur xdelta3-3.1.0.org/xdelta3.c xdelta3-3.1.0/xdelta3.c
--- xdelta3-3.1.0.org/xdelta3.c	2016-01-05 09:05:31.000000000 +0100
+++ xdelta3-3.1.0/xdelta3.c	2026-06-28 16:40:21.210129545 +0200
@@ -1084,7 +1084,7 @@
 #define A32_DO8(buf,i)  A32_DO4(buf,i); A32_DO4(buf,i+4);
 #define A32_DO16(buf)   A32_DO8(buf,0); A32_DO8(buf,8);
 
-static uint32_t adler32 (uint32_t adler, const uint8_t *buf, usize_t len)
+static uint32_t xd3_adler32 (uint32_t adler, const uint8_t *buf, usize_t len)
 {
     uint32_t s1 = adler & 0xffffU;
     uint32_t s2 = (adler >> 16) & 0xffffU;
diff -Naur xdelta3-3.1.0.org/xdelta3.h xdelta3-3.1.0/xdelta3.h
--- xdelta3-3.1.0.org/xdelta3.h	2016-01-08 07:43:21.000000000 +0100
+++ xdelta3-3.1.0/xdelta3.h	2026-06-28 16:21:58.615427914 +0200
@@ -170,16 +170,16 @@
 typedef uint64_t xoff_t;
 /* Note: The following generates benign warnings in a mingw
  * cross-compiler */
-#define Q "I64"
+#define XD3_Q "I64"
 #elif SIZEOF_UNSIGNED_LONG == 8
 typedef unsigned long xoff_t;
-#define Q "l"
+#define XD3_Q "l"
 #elif SIZEOF_SIZE_T == 8
 typedef size_t xoff_t;
-#define Q "z"
+#define XD3_Q "z"
 #elif SIZEOF_UNSIGNED_LONG_LONG == 8
 typedef unsigned long long xoff_t;
-#define Q "ll"
+#define XD3_Q "ll"
 #endif /* typedef and #define Q */
 
 #define SIZEOF_XOFF_T 8
@@ -195,7 +195,7 @@
 #endif /* xoff_t is 32 bits */
 
 #define SIZEOF_XOFF_T 4
-#define Q
+#define XD3_Q
 #endif /* 64 vs 32 bit xoff_t */
 
 /* Settings based on the size of usize_t (32 and 64 bit window size) */
@@ -206,16 +206,16 @@
 typedef uint64_t usize_t;
 /* Note: The following generates benign warnings in a mingw
  * cross-compiler */
-#define W "I64"
+#define XD3_W "I64"
 #elif SIZEOF_UNSIGNED_LONG == 8
 typedef unsigned long usize_t;
-#define W "l"
+#define XD3_W "l"
 #elif SIZEOF_SIZE_T == 8
 typedef size_t usize_t;
-#define W "z"
+#define XD3_W "z"
 #elif SIZEOF_UNSIGNED_LONG_LONG == 8
 typedef unsigned long long usize_t;
-#define W "ll"
+#define XD3_W "ll"
 #endif /* typedef and #define W */
 
 #define SIZEOF_USIZE_T 8
@@ -231,19 +231,19 @@
 #endif /* usize_t is 32 bits */
 
 #define SIZEOF_USIZE_T 4
-#define W
+#define XD3_W
 
 #endif /* 64 vs 32 bit usize_t */
 
 /* Settings based on the size of size_t (the system-provided,
  * usually-but-maybe-not an unsigned type) */
 #if SIZEOF_SIZE_T == 4
-#define Z "z"
+#define XD3_Z "z"
 #elif SIZEOF_SIZE_T == 8
 #ifdef _WIN32
-#define Z "I64"
+#define XD3_Z "I64"
 #else /* !_WIN32 */
-#define Z "z"
+#define XD3_Z "z"
 #endif /* Windows or not */
 #else
 #error Bad configure script
