--- libcdio-2.1.0.org/lib/iso9660/iso9660.c	2019-04-13 17:15:15.000000000 +0200
+++ libcdio-2.1.0/lib/iso9660/iso9660.c	2020-03-24 07:28:19.515232677 +0100
@@ -361,6 +361,8 @@
   from a Unix time structure. timezone is given as an offset
   correction in minutes.
 */
+#pragma GCC push_options
+#pragma GCC diagnostic ignored "-Wformat-truncation"
 void
 iso9660_set_ltime_with_timezone(const struct tm *p_tm,
                                 int time_zone,
@@ -373,7 +375,6 @@

   if (!p_tm) return;

-#pragma GCC diagnostic ignored "-Wformat-truncation"
   snprintf(_pvd_date, 17,
            "%4.4d%2.2d%2.2d" "%2.2d%2.2d%2.2d" "%2.2d",
            p_tm->tm_year + 1900, p_tm->tm_mon + 1, p_tm->tm_mday,
@@ -393,6 +394,7 @@
     pvd_date->lt_gmtoff = 52;
   }
 }
+#pragma GCC pop_options

 /*!
   Set "long" time in format used in ISO 9660 primary volume descriptor
